src/Providers/AuthUserProvider.tsx

```diff
@@ -42,26 +42,43 @@ export default function AuthUserProvider({
     enabled: !!localStorage.getItem(LocalStorageKeys.accessToken),
   });
 
+  const refresh = localStorage.getItem(LocalStorageKeys.refreshToken);
+
+  const { data: refreshTokenData, error: refreshTokenError } = useQuery({
+    queryKey: ["currentUser", accessToken],
+    queryFn: query(routes.token_refresh, { body: { refresh: refresh || "" } }),
+    refetchInterval: careConfig.auth.tokenRefreshInterval,
+    enabled: !!refresh,
+  });
+  if (refreshTokenError) {
+    localStorage.removeItem(LocalStorageKeys.accessToken);
+    localStorage.removeItem(LocalStorageKeys.refreshToken);
+  }
+  if (refreshTokenData) {
+    localStorage.setItem(LocalStorageKeys.accessToken, refreshTokenData.access);
+    localStorage.setItem(
+      LocalStorageKeys.refreshToken,
+      refreshTokenData.refresh,
+    );
+  }
   useEffect(() => {
     if (!user) {
       return;
     }
-
-    updateRefreshToken(true);
-    setInterval(
-      () => updateRefreshToken(),
-      careConfig.auth.tokenRefreshInterval,
-    );
   }, [user]);
 
+  const { mutateAsync: signInMutate } = useMutation({
+    mutationFn: mutate(routes.login),
+  });
```

Comment:
you could move the entire useCallback's def. below to here and that way the `useCallback` would not even be required in the first place

and keep things in `onSuccess` and `onError`.